test/unit/modules/emitter.test.js

```diff
@@ -50,4 +50,59 @@ describe('#emitter', () => {
         assert(emitterInstance.plugins.length === 1, 'plugin works!');
         done();
     });
+
+    it('should get a string state', (done) => {
+        emitterInstance.setState('stringKey', 'hello world');
+        assert(emitterInstance.getState('stringKey') === 'hello world', 'got the expected value');
+        done();
+    });
+
+    it('should get an object state', (done) => {
+        emitterInstance.setState('objectKey', { users: 12, chatName: 'test' });
+        assert.deepEqual(emitterInstance.getState('objectKey'), { users: 12, chatName: 'test' });
+        done();
+    });
+
+    it('should get the list of states', (done) => {
+        emitterInstance.setState('integerKey', 10);
+        emitterInstance.setState('stringKey', 'hello world');
+        emitterInstance.setState('booleanKey', true);
+        assert.deepEqual(emitterInstance.listStates(), ['integerKey', 'stringKey', 'booleanKey']);
+        done();
+    });
+
+    it('should remove a state', (done) => {
+        emitterInstance.setState('integerKey', 10);
+        emitterInstance.setState('integerKey', null);
+
+        assert.isUndefined(emitterInstance.getState('integerKey'));
+        done();
+    });
+
+    it('should manage state from a plugin', (done) => {
+
+        let plugin = () => {
+            class extension {
+                construct() {
+                    this.setState('stringKey', 'plugin state');
```

Comment:
This should set ```Emitter.statePlugin.stringKey```.

Reply:
we can organize the variable of each plugin using the namespace e.g
this.unreadMessages = { isActive: true };

and  from the plugin only call this.set(‘isActive’, false);